### PR TITLE
fby3.5: hd: Initialize HalfDome BIC

### PR DESCRIPTION
**Pull request by Yi-Shum**
Summary:
- Initialize HalfDome BIC.
- Create project config, dts files, file header, and CMake file.
- Add project code in meta-facebook/yv35-hd/.

Test plan:
- Build code: PASS
- Can access BIC console: PASS

LOG:
1. Build code pass.
```
$ west build -b ast1030_evb -p auto ./meta-facebook/yv35-hd/
...
[254/261] Linking C executable zephyr/zephyr_prebuilt.elf

[261/261] Linking C executable zephyr/Y35BHD.elf
Memory region         Used Size  Region Size  %age Used
         SRAM_NC:        152 KB       320 KB     47.50%
           FLASH:          0 GB         0 GB
            SRAM:      385440 B       448 KB     84.02%
        IDT_LIST:          0 GB         2 KB      0.00%
```
2. BIC console:
```
[00:00:00.002,000] <wrn> usb_dc_aspe*** Booting Zephyr OS build v00.01.05-5-g6163ab05ded5  ***
Hello, welcome to yv35 halfdome ffff.ff.ff
```

### CODE REVIEW

File: meta-facebook/yv35-hd/src/platform/plat_ipmb.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#ifndef PLAT_IPMB_H
+#define PLAT_IPMB_H
+
+#include "plat_i2c.h"
+#include "ipmb.h"
+
+#define SELF_I2C_ADDRESS 0x20
+#define MAX_IPMB_IDX 2
+
+extern IPMB_config pal_IPMB_config_table[];
```
Comment (GoldenBug) [inline meta-facebook/yv35-hd/src/platform/plat_ipmb.h:10]: Is this actually needed?

**Review by GoldenBug — CHANGES_REQUESTED** [review]

File: meta-facebook/yv35-hd/src/platform/plat_guid.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#ifndef PLAT_GUID_H
+#define PLAT_GUID_H
+
+enum { MB_SYS_GUID_ID,
+       // OTHER_GUID_ID
+       MAX_GUID_ID,
+};
+
+#endif
```
[the remark this answers is not in the document]
Comment (GoldenBug) [reply meta-facebook/yv35-hd/src/platform/plat_guid.h:9]: You can ignore this comment, I see that it is changed in a later PR.

File: meta-facebook/yv35-hd/prj.conf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+# cmsis-os2
+CONFIG_CMSIS_RTOS_V2=y
+CONFIG_NUM_PREEMPT_PRIORITIES=56
+CONFIG_HEAP_MEM_POOL_SIZE=32768
+CONFIG_CMSIS_V2_MSGQ_MAX_DYNAMIC_SIZE=8192
+CONFIG_POLL=y
+CONFIG_THREAD_NAME=y
+CONFIG_THREAD_STACK_INFO=y
+CONFIG_THREAD_MONITOR=y
+CONFIG_IRQ_OFFLOAD=y
```
[the remark this answers is not in the document]
Comment (Yi-Shum) [reply meta-facebook/yv35-hd/prj.conf:11]: OK, I will remove this config.

File: meta-facebook/yv35-hd/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+set(common_path ../../common)
+FILE(GLOB app_sources src/ipmi/*.c src/platform/*.c src/lib/*.c)
+FILE(GLOB common_sources ${common_path}/service/*.c ${common_path}/service/*/*.c ${common_path}/hal/*.c ${common_path}/dev/*.c ${common_path}/lib/*.c ${common_path}/shell/*.c)
+
+target_sources(app PRIVATE ${app_sources})
+target_sources(app PRIVATE ${common_sources})
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/include/portability)
+target_include_directories(app PRIVATE ${common_path} ${common_path}/service/ipmi/include ${common_path}/service/host ${common_path}/service/sensor ${common_path}/service/usb ${common_path}/service/ipmb ${common_path}/service/mctp ${common_path}/service/pldm ${common_path}/hal ${common_path}/dev/include ${common_path}/lib ${common_path}/shell)
+target_include_directories(app PRIVATE src/ipmi/include src/lib src/platform)
```
[the remark this answers is not in the document]
Comment (Yi-Shum) [reply meta-facebook/yv35-hd/CMakeLists.txt:16]: OK, I'll add this.

**Pull request closed without merging**